diff --git a/envsetup.sh b/envsetup.sh
index 48862f7a66..6bfb7f37de 100644
--- a/envsetup.sh
+++ b/envsetup.sh
@@ -15,6 +15,11 @@
 # gettop is duplicated here and in shell_utils.mk, because it's difficult
 # to find shell_utils.make without it for all the novel ways this file can be
 # sourced.  Other common functions should only be in one place or the other.
+
+
+export device="lilac"
+export release="bp1a"
+
 function _gettop_once
 {
     local TOPFILE=build/make/core/envsetup.mk
